CRM/Activityreport/Data.php

```diff
@@ -15,65 +16,119 @@ class CRM_Activityreport_Data {
    * 
    * @return array
    */
-  public static function get($offset = 0, $limit = 0) {
+  public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0) {
     self::$fields = self::getActivityFields();
     self::$emptyRow = self::getEmptyRow();
     self::$multiValues = array();
+    $cacheKey = 'get-' . $offset . ',' . $limit . ',' . $multiValuesOffset;
 
-    $params = array(
-      'sequential' => 1,
-      'is_current_revision' => 1,
-      'is_deleted' => 0,
-      'api.ActivityContact.get' => array(),
-      'return' => implode(',', array_keys(self::$fields)),
-      'options' => array(
-        'sort' => 'id ASC',
-        'offset' => $offset,
-        'limit' => $limit,
-      ),
-    );
-    $activities = civicrm_api3('Activity', 'get', $params);
+    $result = CRM_Core_BAO_Cache::getItem('activityreport', $cacheKey);
+    if (empty($result)) {
+      $params = array(
+        'sequential' => 1,
+        'is_current_revision' => 1,
+        'is_deleted' => 0,
+        'return' => implode(',', array_keys(self::$fields)),
+        'options' => array(
+          'sort' => 'id ASC',
+          'offset' => $offset,
+          'limit' => $limit,
+        ),
+      );
+      $activities = civicrm_api3('Activity', 'get', $params);
+      $formattedActivities = self::formatResult($activities['values']);
+      $result = self::splitMultiValues($formattedActivities, $offset, $multiValuesOffset);
+      CRM_Core_BAO_Cache::setItem($result, 'activityreport', $cacheKey);
+    }
 
-    return self::splitMultiValues(self::formatResult($activities['values']));
+    return $result;
   }
 
   /**
    * Return an array containing $data rows and each row containing multiple values
    * of at least one field is populated into separate row for each field's
    * multiple value.
    * 
-   * @param array   $data       array containing a set of Activities
+   * @param array   $data               array containing a set of Activities
```

Comment:
I don't think there is a standard followed here , but according to drupal standard the param description should be at new line 

```php
@param int $totalOffset
  Activity absolute offset we start wit
```